File: dp-core/src/main/java/com/dataplatform/data/entity/ChartFolder.java
package com.dataplatform.data.entity;

import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 图表文件夹实体
 */
@Data
public class ChartFolder {
    private Long id;

    /** 文件夹名称 */
    private String folderName;

    /** 父文件夹ID */
    private Long parentId;

    /** 排序 */
    private Integer sortOrder;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 子文件夹（树结构用） */
    private transient List<ChartFolder> children;
}

File: dp-ui/src/api/chartFolder.ts
import request from '@/api/request'

export interface ChartFolder {
  id?: number
  folderName: string
  parentId?: number
  sortOrder?: number
  createTime?: string
  children?: ChartFolder[]
}

/** 获取文件夹树 */
export const getChartFolderTree = () => {
  return request.get('/chart-folder/tree')
}

/** 获取全部列表 */
export const getChartFolderList = () => {
  return request.get('/chart-folder/list')
}

/** 创建文件夹 */
export const createChartFolder = (data: ChartFolder) => {
  return request.post('/chart-folder', data)
}

/** 更新文件夹 */
export const updateChartFolder = (id: number, data: ChartFolder) => {
  return request.put(`/chart-folder/${id}`, data)
}

/** 删除文件夹 */
export const deleteChartFolder = (id: number) => {
  return request.delete(`/chart-folder/${id}`)
}
